Source/scrollrt.cpp

```diff
@@ -225,6 +256,68 @@ static void scrollrt_draw_cursor_item()
 	} else {
 		CelClippedDrawSafe(mx + SCREEN_X, my + cursH + SCREEN_Y - 1, pCursCels, pcurs, cursW);
 	}
+
+	// copy cursor from back buffer to a separate bitmap
+	if (hwcursor) {
+		src = &gpBuffer[SCREENXY(sgdwCursX, sgdwCursY)];
+		dst = sgCursorBitmap;
+		for (i = sgdwCursHgt; i != 0; i--, src += BUFFER_WIDTH, dst += sgdwCursWdt) {
+			memcpy(dst, src, sgdwCursWdt);
+		}
+	}
+
+	// remove cursor from the back buffer
+	if (hwcursor) {
+		src = sgSaveBack;
+		dst = &gpBuffer[SCREENXY(sgdwCursX, sgdwCursY)];
+		for (i = sgdwCursHgt; i != 0; i--, src += sgdwCursWdt, dst += BUFFER_WIDTH) {
+			memcpy(dst, src, sgdwCursWdt);
+		}
+	}
+
+	// update system cursor if the bitmap has changed
+	if (hwcursor) {
+		if (sgCursorCel != pcurs || sgCursorCol != col ||
+			sgCursorPaletteVersion != pal_surface_palette_version || sgCursorWindowsResVer != windowResVer) { 
+			// find the actual transparency mask value, fallback to 255
+			if (pass == 1) {
+				char pal[256];
+
+				memset(pal, 0, sizeof(pal));
+
+				src = sgCursorBitmap;
+				for (i = sgdwCursHgt; i != 0; i--, src += sgdwCursWdt) {
+					for (int j = 0; j < sgdwCursWdt; j++) {
+						pal[src[j]] = 1;
+					}
+				}
+
+				for (i = 0; i < 255; i++) {

```

Comment:
Since `pal` has 256 characters, shouldn't this loop be:
```suggestion
				for (i = 0; i <= 255; i++) {
```
or
```suggestion
				for (i = 0; i < 256; i++) {
```
?

Reply:
No, because the initial value for the colorKey is 255 and we definitely know it's present in the bitmap as we'd just memset the whole bitmap to that.